pkg/pool/connection_pool_test.go

```diff
@@ -22,7 +23,10 @@ func TestGet(t *testing.T) {
 		expectedError error
 	}{
 		{
-			pool:          ConnectionPool{cache: map[string]interface{}{"foo": TestFoo{called: "hoot"}}},
+			pool: ConnectionPool{
```

Comment:
Use the constructor and populate the data using methods.

Reply:
why?

Reply:
I get that you can do that and get the same result, just curious why it should be that way.

Reply:
The test touches `ConnectionPool` internals, which means that when you change the implementation you would need to start adjusting the tests as well. Let's say instead of `map` you use slice, then this test would break without good reason.

Reply:
ok fair. I'll change

Reply:
done